source/docs/hardware-reference/talonfx/improving-performance-with-current-limits.rst

```diff
@@ -0,0 +1,217 @@
+
+Stator current is the output current of the motor and is directly proportional to torque.
+
+.. figure:: images/stator-limit.png
+   :alt: 80 A stator current limit being applied
+
+   *80 A stator current limit taking effect*
+
+Stator current limits are used to restrict the torque output of the motor. This can be used to prevent wheel slip, limit acceleration, or avoid damaging a mechanism when running into a hard stop. Stator current limits also reduce heat buildup in the motor.
+
+Since stator current limits also limit supply current, they are also effective at preventing brownouts when accelerating.
+
+.. dropdown:: Relationship between Supply and Stator Current
+   :open:
+
+   .. note:: This explanation ignores energy loss from heat or other inefficiencies.
+
+   By conservation of energy, power going into the motor must equal power out. Since power is equivalent to voltage times current, :math:`V_{supply} * I_{supply} = V_{stator} * I_{stator}`, where :math:`V_{stator}` is the output voltage of the motor. The duty cycle output of a motor is equivalent to :math:`V_{stator} / V_{supply}`, so the relationship between supply and stator current can be described as :math:`I_{supply} = I_{stator} * duty cycle`.
+
+   As an example, if a motor is applied 100% output (~12 V) and has 80 A of measured stator current, then the supply current will also be 80 A. However, if the motor is applied 50% output (~6 V) and has 80 A of measured stator current, then supply current will only be 40 A.
+
+   This means that stator current limits also effectively limit supply current. Supply current will not exceed a stator current limit and is often significantly lower than stator current.
+
+Supply Limits
+^^^^^^^^^^^^^
+
+.. important:: Supply limits are not functional in torque based control modes. Use the ``PeakForwardTorqueCurrent`` and ``PeakReverseTorqueCurrent`` configs instead.
+
+Supply current is the current drawn from the battery. As a result, limiting supply current can be useful to prevent fuses from resetting and breakers from tripping, as well as to improve the longevity of the battery.
+
+.. tip:: Since stator current limits also limit supply current, it is often not necessary to enable both limits.
+
+When the ``SupplyCurrentThreshold`` has elapsed for ``SupplyTimeThreshold`` amount of time, the supply current limiter will activate and reduce motor output until supply current is within range of the limit.
+
+In the rare case where the robot experiences brownouts despite configuring stator current limits, a supply current limit can also further help avoid brownouts. However, such brownouts are most commonly caused by a bad battery or poor power wiring, so those should be examined first.
+
+How to Budget Limits
```

Comment:
I think we should reorganize this section based on the order of steps teams should actually follow:
- Prevent wheel slip or damaging mechanism
- Limit heat buildup?
- Prevent brownouts
- Prevent breaker trips
- Improve battery longevity